src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -540,3 +540,85 @@ def find_data_dependencies (self, data_xpath):
             self.fail(e)
 
         return ref_list
+
+    """
+    get_module_prefix:   get the prefix of a Yang module
+    input:    name of the Yang module
+    output:   prefix of the Yang module
+    """
+    def get_module_prefix(self, module_name):
+        prefix = ""
+        try:
+            module = self.get_module(module_name)
+        except Exception as e:
+            self.fail(e)
+            return prefix
+        else:
+            return module.prefix()
+
+    """
+    str_to_type:   map string to type of node
+    input:    string
+    output:   type
+    """
+    def str_to_type (self, type_str):
+           mapped_type = {
+                "LY_TYPE_DER":ly.LY_TYPE_DER,
+                "LY_TYPE_BINARY":ly.LY_TYPE_BINARY,
+                "LY_TYPE_BITS":ly.LY_TYPE_BITS,
+                "LY_TYPE_BOOL":ly.LY_TYPE_BOOL,
+                "LY_TYPE_DEC64":ly.LY_TYPE_DEC64,
+                "LY_TYPE_EMPTY":ly.LY_TYPE_EMPTY,
+                "LY_TYPE_ENUM":ly.LY_TYPE_ENUM,
+                "LY_TYPE_IDENT":ly.LY_TYPE_IDENT,
+                "LY_TYPE_INST":ly.LY_TYPE_INST,
+                "LY_TYPE_LEAFREF":ly.LY_TYPE_LEAFREF,
+                "LY_TYPE_STRING":ly.LY_TYPE_STRING,
+                "LY_TYPE_UNION":ly.LY_TYPE_UNION,
+                "LY_TYPE_INT8":ly.LY_TYPE_INT8,
+                "LY_TYPE_UINT8":ly.LY_TYPE_UINT8,
+                "LY_TYPE_INT16":ly.LY_TYPE_INT16,
+                "LY_TYPE_UINT16":ly.LY_TYPE_UINT16,
+                "LY_TYPE_INT32":ly.LY_TYPE_INT32,
+                "LY_TYPE_UINT32":ly.LY_TYPE_UINT32,
+                "LY_TYPE_INT64":ly.LY_TYPE_INT64,
+                "LY_TYPE_UINT64":ly.LY_TYPE_UINT64,
+                "LY_TYPE_UNKNOWN":ly.LY_TYPE_UNKNOWN
+           }
+
+           if type_str not in mapped_type:
+               return ly.LY_TYPE_UNKNOWN
+
+           return mapped_type[type_str]
+
+    def get_data_type (self, schema_xpath):
+        try:
+            schema_node = self.find_schema_node(schema_xpath)
+        except Exception as e:
+            print("get_data_type(): Failed to find schema node from xpath: {}".format(schema_xpath))
+            self.fail(e)
+            return None
+
+        if (schema_node is not None):
+           return schema_node.subtype().type().base()
+
+        return ly.LY_TYPE_UNKNOWN
+
+    """
+    get_leafref_type:   find the type of node that leafref references to
+    input:    data_xpath - xpath of a data node
+    output:   type of the node this leafref references to
+    """
+    def get_leafref_type (self, data_xpath):
```

Comment:
We need to know the type of the leafref to convert data from ConfigDB to YANG. Before conversion, we do not have data_xpath. Is it possible to use schema_xpath or value to leafref here, i.e. 
```
				leaf-list members {
					/* leaf-list members are unique by default */

					type leafref {
						path "/port:sonic-port/port:PORT/port:PORT_LIST/port:port_name";<<<<This value
					}
```

Reply:
Added a new API in the 2nd commit:   get_leafref_type_schema(self, schema_xpath)